src/corecel/sys/ScopedProfiling.cc

```diff
@@ -0,0 +1,78 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/sys/ScopedProfiling.cc
+//! \brief The perfetto implementation of \c ScopedProfiling
+//---------------------------------------------------------------------------//
+#include "ScopedProfiling.hh"
+
+#include "celeritas_config.h"
+#include "corecel/io/Logger.hh"
+
+#include "Environment.hh"
+
+#if CELERITAS_USE_PERFETTO
+#    include <perfetto.h>
+
+#    include "detail/TrackEvent.perfetto.hh"
+#endif
+
+namespace celeritas
+{
+#if !CELER_USE_DEVICE
```

Comment:
The `use_profiling` could probably be combined with the `.cuda.cc` version: we're just using a different profiler. Or maybe at this point we should just merge everything into a single file... thoughts?

Reply:
we can definitely combine `use_profiling`. I think I still like having separate files for the other functions, it makes it easier to find the implementation you're looking for, if anything I'd like to split `use_profiling` in a separate file and remove the `#if CELERITAS_USE_PERFETTO` in the `.cc`